### PR TITLE
Handle TimePicker focus on mount

### PR DESCRIPTION
**Pull request by PerrineBoissieres**
In a similar fashion to TextField, we can now use the `focus` prop on the TimePicker to focus the select when the component is mounted.

Focused style (copied from TextField component):
Capture d’écran 2019-11-08 à 15 27 25

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Unreleased
 
-[...]
+- **[NEW]** `focus` prop on `TimePicker` to focus the select on mount
```
Comment (mcellier) [inline CHANGELOG.md]: Added `focus` prop to `TimePicker` should be sufficient

File: src/timePicker/TimePicker.tsx [hunk 2]
```diff
@@ -2,18 +2,6 @@ import React, { PureComponent } from 'react'
 import cc from 'classcat'
 import prefix from '_utils'
 
-/**
- * Whether or not Date#toLocaleTimeString supports arguments `locales` et `options`.
- */
-const toLocaleTimeStringSupportsLocales = (() => {
```
Comment (mcellier) [inline src/timePicker/TimePicker.tsx]: Not used?

File: src/timePicker/TimePicker.tsx [hunk 3]
```diff
@@ -105,6 +96,12 @@ export default class TimePicker extends PureComponent<TimePickerProps, TimePicke
     this.props.onChange({ name, value })
   }
 
+  componentDidMount() {
```
Comment (mcellier) [inline src/timePicker/TimePicker.tsx]: Do we have a specific style when focussed?

File: src/timePicker/TimePicker.tsx (hunk 2 above)
Comment (PerrineBoissieres) [reply src/timePicker/TimePicker.tsx]: Seems not, it was a raised warning by the linter.

File: src/timePicker/TimePicker.tsx (hunk 3 above)
Comment (PerrineBoissieres) [reply src/timePicker/TimePicker.tsx]: I'm adding it (I'll update the description)

**Pull request merged**